strider/trapi_throttle/throttle.py

```diff
@@ -266,43 +266,53 @@ async def process_batch(
-                    # Make a copy
-                    response_values[request_id] = ReasonerResponse(message=Message())
-                    response_values[
-                        request_id
-                    ].message.query_graph = request_value_mapping[
-                        request_id
-                    ].message.query_graph.copy()
-                    response_values[request_id].message.knowledge_graph = (
-                        message.knowledge_graph or KnowledgeGraph(nodes={}, edges={})
-                    ).copy()
-                    response_values[request_id].message.results = (
-                        message.results or HashableSet(__root__=[])
-                    ).copy()
-                else:
-                    # Split using the request_curie_mapping
-                    for request_id, curie_mapping in request_curie_mapping.items():
-                        try:
+                try:
+                    if len(request_curie_mapping) == 1:
+                        request_id = next(iter(request_curie_mapping))
+                        # Make a copy
+                        response_values[request_id] = ReasonerResponse(
+                            message=Message()
+                        )
+                        response_values[
+                            request_id
+                        ].message.query_graph = request_value_mapping[
+                            request_id
+                        ].message.query_graph.copy()
+                        response_values[request_id].message.knowledge_graph = (
+                            message.knowledge_graph
+                            or KnowledgeGraph(nodes={}, edges={})
+                        ).copy()
+                        response_values[request_id].message.results = (
+                            message.results or HashableSet(__root__=[])
+                        ).copy()
+                    else:
+                        # Split using the request_curie_mapping
+                        for request_id, curie_mapping in request_curie_mapping.items():
                             filtered_msg = filter_by_curie_mapping(
                                 message, curie_mapping, kp_id=self.id
                             )
-                            filtered_msg.query_graph = QueryGraph.parse_obj(
-                                request_value_mapping[request_id]
-                            )
+                            filtered_msg.query_graph = request_value_mapping[
+                                request_id
+                            ].message.query_graph.copy()
                             response_values[request_id] = ReasonerResponse(
                                 message=filtered_msg
                             )
-                        except BatchingError as err:
-                            # the response is probably malformed
-                            response_values[request_id] = err
```

Comment:
Just leaving a note that this line is the critical line. As this was previously written it was possible to error out with a different exception. If you don't call this line you get stuck in an infinite while. In the edits here the try is restructured so that we will always fall to the except below for every exception.